packages/terra-table/src/SelectableUtils.js

```diff
@@ -1,7 +1,20 @@
 /**
-* Returns the index of the first selected row for SingleSelectableRows.
-* To be used in the constructor, to set initial state.
-*/
+ * Returns a validated max count for selection based on the rows and max provided.
+ */
+const validatedMaxCountSelection = (rows, maxSelectionCount) => {
+  if (maxSelectionCount !== undefined) {
+    return maxSelectionCount;
+  }
+  if (rows) {
+    return rows.length;
+  }
+  return 0;
+};
```

Comment:
Since rows are actually React children, we can use `return React.Children.count(rows);`